nc_time_axis/__init__.py

```diff
@@ -45,43 +45,36 @@ def __repr__(self):
         )
 
 
+_RESOLUTION_TO_FORMAT = {
+    "SECONDLY": "%H:%M:%S",
+    "MINUTELY": "%H:%M",
+    "HOURLY": "%Y-%m-%d %H:%M",
+    "DAILY": "%Y-%m-%d",
+    "MONTHLY": "%Y-%m",
+    "YEARLY": "%Y"
+}
+
+
 class NetCDFTimeDateFormatter(mticker.Formatter):
     """
     Formatter for cftime.datetime data.
 
     """
-
-    # Some magic numbers. These seem to work pretty well.
-    format_options = [
-        FormatOption(0.0, 0.2, "%H:%M:%S"),
-        FormatOption(0.2, 0.8, "%H:%M"),
-        FormatOption(0.8, 15, "%Y-%m-%d %H:%M"),
-        FormatOption(15, 90, "%Y-%m-%d"),
-        FormatOption(90, 900, "%Y-%m"),
-        FormatOption(900, 6000000, "%Y"),
-    ]
-
     def __init__(self, locator, calendar, time_units):
         #: The locator associated with this formatter. This is used to get hold
         #: of the scaling information.
         self.locator = locator
         self.calendar = calendar
         self.time_units = time_units
 
-    def pick_format(self, ndays):
-        """
-        Returns a format string for an interval of the given number of days.
-
-        """
-        for option in self.format_options:
-            if option.lower < ndays <= option.upper:
-                return option.format_string
-        else:
-            emsg = f"No formatter found for an interval of {ndays} days."
-            raise ValueError(emsg)
+    def pick_format(self, resolution):
+        return _RESOLUTION_TO_FORMAT[resolution]
 
     def __call__(self, x, pos=0):
-        format_string = self.pick_format(ndays=self.locator.ndays)
+        # Note that self.locator.tick_values must be called
+        # before calling this method; otherwise the locator's
+        # resolution attribute will not be defined.
+        format_string = self.pick_format(self.locator.resolution)
```

Comment:
There is an awkward bit of coupling between the `NetCDFTimeDateFormatter` class and the `NetCDFTimeDateLocator` class in that the formatter's `__call__` method assumes that the locator already has a `resolution` attribute assigned to it.  The `resolution` attribute can only be assigned during the `compute_resolution`  method, because it depends on the range of the axis, which is not known a priori. Previously this coupling occurred through the `ndays` attribute that I've now removed, but again was a parameter that needed to be set in the `tick_values` method of the locator due to its dependence on the axis range.  

All that is to say, it feels odd to depend on an attribute that doesn't always exist, but I am following a precedent that was set in the previous implementation, so it appears to be safe.